client/elements/sc-page-dictionary.html

```diff
@@ -268,42 +272,54 @@ <h3>Similar Spelling</h3>
 
           // Adding Adjacent Terms
           _computeAdjacentUrl() {
-              return `${this.$.meta.byKey('API_ROOT')}/adjacent`;
+              return `${this.$.meta.byKey('API_ROOT')}/dictionary_adjacent/${this.dictionaryWord}`;
           }
 
           _loadAdjacent() {
               this.$.adjacent_ajax.url = this._computeAdjacentUrl();
               return this.$.adjacent_ajax.generateRequest();
           }
 
-          _adjacentLoaded() {
-              console.log("loaded Adjacent");
-              console.log(this.AdjacentReturns);
-          }
-
-          _calculateClass(item) {
-              return (item === this.dictionaryWord) ? 'selected-terms-item' : '';
+          // Adding Similar Terms
+          _computeSimilarUrl() {
+              return `${this.$.meta.byKey('API_ROOT')}/dictionary_similar/${this.dictionaryWord}`;
           }
 
-          // Adding glossary results to related items drawer
-          _computeGlossaryUrl() {
-              return `${this.$.meta.byKey('API_ROOT')}/glossary`;
+          _loadSimilar() {
+              this.$.similar_ajax.url = this._computeSimilarUrl();
+              return this.$.similar_ajax.generateRequest();
           }
 
-          _loadGlossary() {
-              this.$.glossary_ajax.url = this._computeGlossaryUrl();
-              return this.$.glossary_ajax.generateRequest();
+          // Calculating if the current item is selected
+          _calculateClass(item) {
+              return (item === this.dictionaryWord) ? 'selected-terms-item' : '';
           }
 
+          // Converts glossary output to object
           _glossaryLoaded() {
-              console.log("loaded Glossary");
-              console.log(this._getGlossaryDescription("dhamma"));
-              console.log(this.glossaryReturns);
+              let result = {};
+              this.glossaryReturns[0].forEach((item) => result[item.word] = item.text);
+              this.glossaryList = result;
           }
 
-          _getGlossaryDescription(glossword) {
-              console.log(this.glossaryReturns(glossword));
-              return this.glossaryReturns(glossword);
+          _getGlossaryItems(glossaryList, inputArray, adjacent=false) {
+              if (inputArray) {
+                  let glossary = [];
+                  let glossaryObject = {};
+                  let glossText = '';
+                  if (glossaryList) {
+                      for (let glossWord in inputArray[0]) {
```

Comment:
is `inputArray` (either `adjacentReturns` or `similarReturns`) always a one-element array? That is, there will never be a `inputArray[1]`? If so, the arangoDB query should probably return the single object instead of a single-object array.

Reply:
I don't quite understand Arango DB in this. It gives me back an array with 10 items inside an array. So the first array has only 1 item. But when I try to tell Arango to give me back Array[0] instead, it gives me a completely blank array. How to do this?

Reply:
I just tried - yeah, it's quite weird. Don't worry about it.